types/src/v2/params.rs

```diff
@@ -104,23 +104,35 @@ impl<'a> RpcParams<'a> {
 			Some(ref cow) => cow,
 			None => return false,
 		};
-
-		json.trim_start().starts_with('{')
+		json.starts_with('{')
 	}
 
 	/// Obtain a sequence parser, [`RpcParamsSequence`].
 	///
 	/// This allows sequential parsing of the incoming params, using an `Iterator`-style API and is useful when the RPC
 	/// request has optional parameters at the tail that may or may not be present.
+	///
+	/// NOTE: there is an edge-case in JSON-RPC context where you would want to
+	/// decode `[1]` as either `Vec<u64>` or `u64` depending on specific method.
+	/// Thus, we decode `[1]` as `Vec<u64>` and not as `u64`.
+	///
+	/// If you need specific behaviour use `RpcParams::parse` or `RpcParams::one`.
 	pub fn sequence(&self) -> RpcParamsSequence {
-		RpcParamsSequence(self.0.as_ref().map(AsRef::as_ref).unwrap_or(""))
+		let json = match self.0.as_ref() {
+			// It's assumed that params is `[a,b,c]`, if empty regard as no params.
+			Some(json) if json == "[]" => "",
```

Comment:
Aah, so this is basically done because by the looks of it, `RpcParamsSequence` doesn't properly handle "[]" as an empty sequence if I understand right (it tries to parse "]" and errors, or something like that).

I guess this is essentially the constructor for `RpcParamsSequence`; something inmy brain would prefer this logic to be on that type, but yeah, it's probably a bit nitpicky :)

Reply:
indeed :)